### ISSUE TITLE
Task: develop tooling to evaluate public/private API usage of a napari-using package

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#3622]
## 🧰 Task
As more and more packages start to build on napari  (regardless of whether they implement a hookspec or entry_points, or just use napari) it would be nice to have a better sense for what the heavily used public API surface is.  In parallel, it would be nice to have a metric to gauge how much of napari's private internals a given plugin is using.  @royerloic suggested that we could perhaps build some tooling that could analyze a package.  

Goals would be:
- encourage best practices of using only public interfaces, making it easier for napari to iterate on its internals
- identify places where public APIs are lacking (i.e. if a number of packages are all using a particular private interface, maybe it's because we really ought to be providing something)
- get a better sense for the "active" API surface out there.  (it's one thing to say "everything that doesn't have an underscore is public", and vice versa... but it'd be even better to start to be more explicit about napari's _supported_ public API, which could effectively have it's own interface version independent of napari's application version).

this could take the form of some AST-based static analysis, or dynamic analysis.  But it would essentially be a script that looks at a package (could download from pypi or github, etc...) and pumps out a report of some sort about API usage.
